radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -478,79 +453,39 @@ public Result transferTokens(
-		final UniqueProperty uniqueProperty;
-		if (unique != null) {
-			// Unique Property must be the from address so that all validation occurs in a single shard.
-			// Once multi-shard validation is implemented this constraint can be removed.
-			uniqueProperty = new UniqueProperty(unique, from);
-		} else {
-			uniqueProperty = null;
-		}
 
-		return executeTransaction(transferTokensAction, null, null, null, null, uniqueProperty);
+		return execute(transferTokensAction);
 	}
 
-	public Single<UnsignedAtom> mapToAtom(
-		@Nullable TransferTokensAction transferTokensAction,
-		@Nullable SendMessageAction sendMessageAction,
-		@Nullable CreateTokenAction tokenCreation,
-		@Nullable MintTokensAction mintTokensAction,
-		@Nullable BurnTokensAction burnTokensAction,
-		@Nullable UniqueProperty uniqueProperty
-	) {
-		if (transferTokensAction != null) {
-			pull(transferTokensAction.getFrom());
-		}
-
-		if (burnTokensAction != null) {
-			pull(burnTokensAction.getTokenClassReference().getAddress());
-		}
-
-		return Observable.concatArray(
-			//				Observable.just(uniquePropertyTranslator.map(uniqueProperty)),
-			transferTokensAction != null ? tokenBalanceStore.getState(transferTokensAction.getFrom())
-				.firstOrError().toObservable()
-				.map(s -> transferTokensToParticlesMapper.map(transferTokensAction, s)) : Observable.empty(),
-			Observable.just(sendMessageToParticlesMapper.map(sendMessageAction)),
-			Observable.just(tokenMapper.map(tokenCreation)),
-			Observable.just(mintTokensActionMapper.map(mintTokensAction)),
-			burnTokensAction != null ? tokenBalanceStore.getState(burnTokensAction.getTokenClassReference().getAddress())
-				.firstOrError().toObservable()
-				.map(s -> burnTokensActionMapper.map(burnTokensAction, s)) : Observable.empty(),
-			Observable.just(
-				Collections.singletonList(SpunParticle.up(new TimestampParticle(System.currentTimeMillis())))
-
-			)
+	/**
+	 * Builds an unsigned atom given a user action. Note that this is
+	 * method will always return a unique atom even if given the equivalent actions
+	 *
+	 * @param action the user action to translate an atom from
+	 * @return the constructed atom
+	 */
+	public Single<UnsignedAtom> buildAtom(Action action) {
+		return Observable.concat(
+			this.actionToParticlesMappers.stream()
+				.map(m -> m.map(action))
+				.collect(Collectors.toList())
 		)
-			.<List<SpunParticle>>scanWith(
-				ArrayList::new,
-				(a, b) -> Stream.concat(a.stream(), b.stream()).collect(Collectors.toList())
-			)
-			.lastOrError()
-			.map(particles -> {
-				List<SpunParticle> allParticles = new ArrayList<>(particles);
-				allParticles.addAll(feeMapper.map(particles, universe, getMyPublicKey()));
-				return allParticles;
-			})
-			.map(particles -> new UnsignedAtom(new Atom(particles)));
+		.concatWith(Observable.just(SpunParticle.up(new TimestampParticle(System.currentTimeMillis()))))
+		.<List<SpunParticle>>scanWith(
```

Comment:
I'm still seeing a lot of `SpunParticle` references in code without generic arguments.  If we really don't want to have to specify generic arguments for `SpunParticle`, then can we make the class definition non-generic?

This seems quite a poor choice to me that we are giving the compiler type hints for how things should go together, and then just ignoring the many type warnings that result from not using these type hints properly.

Reply:
Yup, agreed. Will create a task to specifically look into the best way to resolve this.